### ISSUE TITLE
OSX agent installation script fails when using GNU sed

### ISSUE THREAD

**Issue by mlauter** [DataDog/datadog-agent#15474]
`install_mac_os.sh` script only partially accounts for the fact that mac users could be using GNU sed, causing script execution to fail.

**Agent Environment**
`install_mac_os.sh` on current HEAD of main (SHA: `a4d53960882c1a33012af04af2bbcf505c9276e9`)

**Describe what happened:**

Running the `install_mac_os.sh` script fails on line [242](https://github.com/DataDog/datadog-agent/blob/a4d53960882c1a33012af04af2bbcf505c9276e9/cmd/agent/install_mac_os.sh#L242) when using GNU sed.

Note: The script handles the possibility of encountering gnu sed in the [`new_config` function ](https://github.com/DataDog/datadog-agent/blob/a4d53960882c1a33012af04af2bbcf505c9276e9/cmd/agent/install_mac_os.sh#L205-L215) but not in the [`plist_modify_user_group` function](https://github.com/DataDog/datadog-agent/blob/a4d53960882c1a33012af04af2bbcf505c9276e9/cmd/agent/install_mac_os.sh#L242).

I tried running the `sed -i .backup...` command alone locally -- it does not work for GNU sed. The space between the `-i` flag and the suffix must be removed: `sed -i.backup` works.

**Describe what you expected:**

Script should run to completion and exit cleanly.

**Suggested fix**

Assuming this works for OSX sed, delete the space between `-i` and `.backup`. If not, define the `-i` command conditionally. If this sounds reasonable I can open a PR.

**Steps to reproduce the issue:**

1. Install `gnu-sed` via homebrew and add a gnubin dir to your path (https://formulae.brew.sh/formula/gnu-sed). 
2. run the install script

**Additional environment details (Operating System, Cloud provider, etc):**
OSX 13.1

**Comment by bkabrda** [comment]
Thanks a lot for the report @mlauter, I'll take a look and submit a fix for this.

**Comment by mlauter** [comment]
@bkabrda Opened one just now because why not, but totally fine if you had something else in mind!

**Comment by bkabrda** [comment]
@mlauter thanks for doing it. I opened another one where I try to make handling of GNU sed a little more systematic (plus also the backupfile is not really necessary in this case, so I just removed it): https://github.com/DataDog/datadog-agent/pull/15694

Sorry I didn't notice your PR earlier, GH failed to refresh this page. I'd propose going forward with my PR because it creates the function that will help us in future sed uses as well. WDYT?
